lib/flight_scheduler/job_registry.rb

```diff
@@ -70,14 +70,14 @@ def remove_old_jobs
         begin
           # Its possible that we're still processing the request to deallocate
           # the job.  If that's the case, we'll clean this job up another time.
-          if job.allocated?
+          if job.allocated? || FlightScheduler.app.job_registry.tasks_for(job).any?(&:allocated?)
```

Comment:
This is quite an esoteric conditional. I assume that if an `array_job` has been allocated?

If so, its probably best to roll it into `job.allocated?`. Something like:

```
def allocated?
  if type == 'ARRAY_JOB'
    FlightScheduler.app.job_registry.tasks_for(job).any?(&:allocated?)
  else
     ...
  end
end
```

Reply:
I thought about doing that, but I decided against it.  I don't want to end up in a situation where `job.allocated? == true` but `job.allocation == nil`.  That seems worse than this conditional.  Also, `job.allocation` returning an array of allocations does not seem right.  Perhaps a `job.allocated?(include_tasks: true)` would work?